src/Sarif/Core/Stack.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using System.Text;
+using DotNetStackFrame = System.Diagnostics.StackFrame;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public enum StackFormat
+    {
+        Default,
+        TrailingNewLine
+    }
+
+    /// <summary>
+    /// A call stack relevant to a SARIF result.
+    /// </summary>
+    public partial class Stack
+    {
+        private const string IN = " in ";
+        private const string AT = "   at ";
+        private const string LINE = ":line";
+
+        /// <summary>
+        /// Creates a SARIF Stack instance from a .NET StackTrace instance
+        /// </summary>
+        /// <param name="stackTrace"></param>
+        /// <returns></returns>
+        public static Stack Create(StackTrace stackTrace)
+        {
+            Stack stack = new Stack();
+
+            if (stackTrace.FrameCount == 0) { return stack; }
+
+            stack.Frames = new StackFrame[stackTrace.FrameCount];
+
+            for (int i = 0; i < stackTrace.FrameCount; i++)
+            {
+                stack.Frames[i] = StackFrame.Create(stackTrace.GetFrame(i));                
+            }
+            return stack;
+        }
+
+        public static IEnumerable<Stack> Create(Exception exception)
+        {
+            List<Stack> stacks;
+            Queue<Exception> exceptions;
+
+            stacks = new List<Stack>();
+            exceptions = new Queue<Exception>(new Exception[] { exception });
+
+            while (exceptions.Count > 0)
+            {
+                Stack stack;
+                Exception current;
+                
+                current = exceptions.Dequeue();
+
+                if (current.InnerException != null)
+                {
+                    exceptions.Enqueue(current.InnerException);
+                }
+
+                stack = Create(current.StackTrace);
+
+                stack.Message = current.Message;
+
+                stacks.Add(stack);
+            }
+
+            return stacks;
+        }
+
+        public static Stack Create(string stackTrace)
+        {
```

Comment:
So the non-.NET-specific stack creation: e.g. building up from list of plain SARIF locations is not shown here because it is generated from schema, right? I'd need to look at the full API as it shows in IntelliSense, but maybe this method would benefit from a better name (though I can't come up with anything pretty -- CreatFromDotNetString? yuck)  to leave room for helpers from other well-known formats from other environments? Or maybe have a StackFormat input arg and we can have StackFormat.DotNet?

I also wonder if we shouldn't be careful about adding signifcant dependencies directly as members of the core schema OM. e.g. Should helpers like this go in a different assembly from the pure OM? That would be my BCL instinct, but maybe it's not worth the complexity here. Also, do we have a task to make the SARIF SDK a portable class library? It will definitely need to work in .NET Core for x-plat tools and it's better to do early than late when you may have taken too many dependencies. (I thought for a minute that StackTrace had been gutted too much for this in .NET Core, which prompted this line of thinking, but looking at corefx repo now and I think this is OK.  Still, it would be better to let the compilation handle future portability rather than just thinking about it.)